File: app/src/main/java/com/mlsdev/mlsdevstore/data/local/database/AppDatabase.java
package com.mlsdev.mlsdevstore.data.local.database;

import android.arch.persistence.room.Database;
import android.arch.persistence.room.RoomDatabase;
import android.arch.persistence.room.TypeConverters;

import com.mlsdev.mlsdevstore.data.local.database.converter.CategoryConverter;
import com.mlsdev.mlsdevstore.data.local.database.dao.CategoriesDao;
import com.mlsdev.mlsdevstore.data.model.category.Category;
import com.mlsdev.mlsdevstore.data.model.category.CategoryTree;
import com.mlsdev.mlsdevstore.data.model.category.CategoryTreeNode;
import com.mlsdev.mlsdevstore.data.model.user.Address;
import com.mlsdev.mlsdevstore.data.model.user.CreditCard;
import com.mlsdev.mlsdevstore.data.model.user.PersonalInfo;

@Database(entities = {
        Category.class,
        CategoryTreeNode.class,
        CategoryTree.class,
        Address.class,
        CreditCard.class,
        PersonalInfo.class
}, version = 1, exportSchema = false)
@TypeConverters({CategoryConverter.class})
public abstract class AppDatabase extends RoomDatabase {
    public abstract CategoriesDao categoriesDao();
}

File: app/src/main/java/com/mlsdev/mlsdevstore/data/local/database/AppDatabase.kt
package com.mlsdev.mlsdevstore.data.local.database

import androidx.room.Database
import androidx.room.RoomDatabase
import androidx.room.TypeConverters
import com.mlsdev.mlsdevstore.data.local.database.converter.CategoryConverter
import com.mlsdev.mlsdevstore.data.local.database.dao.*
import com.mlsdev.mlsdevstore.data.model.category.CategoryTree
import com.mlsdev.mlsdevstore.data.model.category.CategoryTreeNode
import com.mlsdev.mlsdevstore.data.model.image.CategoryImageEntity
import com.mlsdev.mlsdevstore.data.model.order.GuestCheckoutSession
import com.mlsdev.mlsdevstore.data.model.product.Product
import com.mlsdev.mlsdevstore.data.model.user.Address
import com.mlsdev.mlsdevstore.data.model.user.PersonalInfo

@Database(entities = [
    CategoryTreeNode::class,
    CategoryTree::class,
    Address::class,
    PersonalInfo::class,
    GuestCheckoutSession::class,
    CategoryImageEntity::class,
    Product::class
], version = 1, exportSchema = false)
@TypeConverters(CategoryConverter::class)
abstract class AppDatabase : RoomDatabase() {
    abstract fun categoriesDao(): CategoriesDao
    abstract fun addressDao(): AddressDao
    abstract fun personalInfoDao(): PersonalInfoDao
    abstract fun checkoutSessionDao(): CheckoutSessionDao
    abstract fun categoryImagesDao(): CategoryImagesDao
    abstract fun productsDao(): ProductsDao
}
